Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector.py

```diff
@@ -0,0 +1,403 @@
+
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC
+VENDOR = '1Password'
+PRODUCT = '1Password'
+
+# Default results per page (optimal value)
+# > 1000 causes HTTP 400 [Bad Request]
+# < 1000 increases the number of requests and may eventually trigger HTTP 429 [Rate Limit]
+DEFAULT_RESULTS_PER_PAGE = 1000
+
+DEFAULT_MAX_EVENTS_PER_FETCH = 1000
+
+DEFAULT_FETCH_FROM_DATE = datetime.now(tz=UTC) - timedelta(weeks=2)  # 2 weeks ago in UTC timezone
+
+EVENT_TYPE_MAPPING = CaseInsensitiveDict(
+    {
+        # Display name: API Feature and endpoint name
+        'Item usage actions': 'itemusages',
+        'Audit events': 'auditevents',
+        'Sign in attempts': 'signinattempts',
+    }
+)
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the 1Password Events API"""
+
+    def introspect(self) -> dict:
+        """Performs an introspect API call to check authentication status and the features (scopes) that the token can access,
+        including one or more of: 'auditevents', 'itemusage', and 'signinattempts'.
+
+        Raises:
+            DemistoException: If API responds with an HTTP error status code.
+
+        Returns:
+            dict: The response JSON from the 'Auth introspect' endpoint.
+        """
+        return self._http_request(method='GET', url_suffix='/auth/introspect', raise_on_status=True)
+
+    def get_events(
+        self,
+        event_type: str,
+        from_date: datetime | None = None,
+        results_per_page: int = DEFAULT_RESULTS_PER_PAGE,
+        pagination_cursor: str | None = None,
+    ):
+        """Gets events from 1Password based on the specified event type
+
+        Args:
+            event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+            from_date (datetime | None): Optional datetime from which to get events.
+            results_per_page (int): The maximum number of records in response (Recommended to use default value).
+            pagination_cursor (str | None): Pagination Cursor from previous API response.
+
+        Raises:
+            ValueError: If unknown event type or missing cursor and start time.
+            DemistoException: If API responds with an HTTP error status code.
+
+        Returns:
+            dict: The response JSON from the event endpoint.
+        """
+        if not (feature := EVENT_TYPE_MAPPING.get(event_type)):
+            raise ValueError(f'Invalid or unsupported 1Password event type: {event_type}.')
+
+        body: dict[str, Any]
+        if pagination_cursor:
+            demisto.debug(f'Requesting events of type: {event_type} using pagination cursor: {pagination_cursor}')
+            body = {'cursor': pagination_cursor}
+
+        elif from_date:
+            formatted_from_date: str = from_date.strftime(DATE_FORMAT)
+            demisto.debug(f'Requesting events of type: {event_type} using from date: {formatted_from_date}')
```

Comment:
This is no longer relevant since the client method has changed to contain less logic (As we agreed)

A similar log message has been added to the new client method.